### PR TITLE
feat(route): add 小黑盒喜加一

### PR DESCRIPTION
**Pull request by ladeng07**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/xiaoheihe/add2cart/epic
/xiaoheihe/add2cart/steam
/xiaoheihe/add2cart/gog
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/xiaoheihe/add2cart.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+async function handler(ctx) {
+    const platform = ctx.req.param('platform');
+    const response = await got(`https://api.xiaoheihe.cn/mall/add_to_cart/?platform=${platform}`);
+    const data = response.data.result.games;
+
+    let items = data.map((item) => {
+        const title = `${item.type === 'dlc' ? '[DLC]' : ''}${item.name}${item.name_en ? '/' + item.name_en : ''}`;
+
+        let description = `<img src="${item.image}"/> <br/>`;
+        description += String(item.type === 'dlc' ? '本体: ' + item.product_home_name + '<br/>' : '');
```
Comment (TonyRL) [inline lib/routes/xiaoheihe/add2cart.ts:24]:
```suggestion
        description += item.type === 'dlc' ? '本体: ' + item.product_home_name + '<br/>' : '';
```

File: lib/routes/xiaoheihe/add2cart.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        if (item.price) {
+            description += `原价: ${item.price.initial_amount}<br/>`;
+        }
+        if (item.score) {
+            description += `评分: ${item.score}<br/>`;
+        }
+
+        description += `支持中文: ${item.chinese_support ? '是' : '否'}<br/>`;
+
+        description += `截止时间: ${new Date(item.end_time * 1000 + 8 * 3600 * 1000).toLocaleString('zh-CN')}<br/>`;
```
Comment (TonyRL) [inline lib/routes/xiaoheihe/add2cart.ts:34]: `item.end_time` is a Unix timestamp that does not require timezone adjustment.

File: lib/routes/xiaoheihe/add2cart.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            link,
+            pubDate: parseDate(item.end_time * 1000),
+        };
+    });
+
+    if (items.length === 0) {
+        items = [
+            {
+                title: `${platform.toUpperCase()}最近没有喜加一(悲`,
+                pubDate: parseDate(response.data.result.weixindata.timestamp * 1000).setHours(-15, 0, 0, 0),
```
Comment (TonyRL) [inline lib/routes/xiaoheihe/add2cart.ts:53]: `response.data.result.weixindata.timestamp` is a Unix timestamp that does not require timezone adjustment.

File: lib/routes/xiaoheihe/add2cart.ts (hunk 2 above)
Comment (ladeng07) [reply lib/routes/xiaoheihe/add2cart.ts:34]: `item.end_time` appears in the body content and does not automatically convert time zones, so manual conversion is required.

**Pull request merged**